File: se_random/ObStack.java
//package iProlog;
import java.util.ArrayList;

class ObStack<T> extends ArrayList<T> {
public static int[] a = new int[15];

  private static final long serialVersionUID = 1L;

  final T pop() {
    final int last = this.size() - 1;
    return this.remove(last);
  }

  final void push(final T O) {
    add(O);
  }

  final T peek() {
    return get(this.size() - 1);
  }

public static void random(int k)
{
int f= 435,b=786,m=85;
a[0]=6;
int i=0;
while(i<10)
{
int result=((a[i]*f)*b)%m;
i++;
a[i]=result;
}
}


public static void main(final String[] args) {
random(6);
    ObStack s=new ObStack();
s.push(a[1]);
System.out.println(s.peek());
s.push(a[2]);


System.out.println(s.peek());

  }

}

File: se_random/ObStack.cpp
#include<iostream>
#include <vector>
class random;
int a[15];

template<typename T>
	class ObStack : public std::vector<T>
	{

  private:
	//  static long serialVersionUID = 1LL;

  public:
	  T pop()
	  {
		int last = this->size() - 1;
		return this->remove(last);
	  }

	  void push(T const O)
	  {
		this->push_back(O);
	  }

	  T peek()
	  {
		return this->at(this->size() - 1);
	  }
	};

int main()
{
int f= 435,b=786,m=85,i=0;
a[0]=6;
while(i<10)
{
int result=((a[i]*f)*b)%m;
i++;
a[i]=result;
}

ObStack <int>o;
o.push(a[1]);
std::cout<<o.peek()<<std::endl;
o.push(a[2]);
std::cout<<o.peek()<<std::endl;
}